### PR TITLE
New layout

### PR DESCRIPTION
**Pull request by kojoty**
This PR contains the proposition of the new layout - this is not ready yet - this is only an early stage of works. This not affect current pages - it can be seen by test.php only. There is still lack of menu-handling code - I'm working on it now.

Like I wrote in emails - the main idea is to create the similar layout to current common/main.tpl.php but with bootstrap css. This is a step on a way for further responsive design. We will be able to migrate pages to the bootstrap-like styles.

This is little in context of  #1075. 

@deg-pl, @andrixnet, @harrieklomp: 
I need config change (before commit this PR):
please add to your settings.inc.php line: `$config['ocNode'] = "<your-node-id>";`
For example for OCPL it should be `$config['ocNode'] = "pl";` 'ro' for OCRO, and so on...

It is replacement of node detection by domain which was used before in many places in the code.

If you want to look at this there is also my proposition of the "new" template's subsystem. TestController doesn't use old templates and eval - there are two simple methods which allow to build view-template file. Template contains only pure php injections to html. All values used in template have to be injected by $view->var, translations are done by $tr() function - everything is (in my opinion simple, clear and useful) - the only disadvantages is that var injection needs a little more chars - <?=...?> instead of {} - but there is huge advantage of this solution - there is no need of any parsing and rewriting of template file - this is just php file which is just included in the right place and time. Please take a look.

### CODE REVIEW

**Comment by harrieklomp** [comment]
Add new lines on settings.inc.php on OCNL

File: Utils/View/View.php [hunk 1]
```diff
@@ -184,23 +202,70 @@ public function addLocalCss($css_url){
...
+    {
         //TODO: refactoring needed but this is still this way
         tpl_set_tplname($tplName);
+        $this->_template = $tplName;
     }
 
-    public function buildView(){
-        //TODO: refactoring needed but this is still this way
+    /**
+     * Wrapper for obsolate template system.
```
Comment (deg-pl) [inline Utils/View/View.php]: 3 szyszki. Choć za komentarz mogą być 2. (obsolete)

File: index.php [hunk 2]
```diff
@@ -41,8 +41,7 @@
 include ($dynstylepath . "totalstats.inc.php");
 
 // diffrent oc server handling: display proper info depend on server running the code
-$nodeDetect = substr($absolute_server_URI, - 3, 2);
```
Comment (deg-pl) [inline index.php]: :+1:

File: lib/settingsDefault.inc.php [hunk 3]
```diff
@@ -289,3 +295,19 @@
 
 $subject_prefix_for_site_mails = "OCXX";
 $subject_prefix_for_reviewers_mails = "";
+
+
+// keywords in every header
+$config['header']['keywords'] = 'geocaching, opencaching, skarby,'.
```
Comment (deg-pl) [inline lib/settingsDefault.inc.php]: :+1: Very good idea!

**Comment by deg-pl** [comment]
1) OC PL and OC UK ready (have $config['ocNode'])
2) What is preffered $config['ocNode'] setting on devel site?
3) Does this PR fix #958 and #1137?
4) Can we remove "oc_on_all_pages_top_1A" and "oc_subtitle_on_all_pages_1A" from translation files?
5) Can we remove nonlusers.txt after merge?

**Comment by deg-pl** [comment]
6. Could you also change meta name="keywords" to new config value - in main.tpl.php mini.tpl.php

In mini.tpl.php we have not HTML5 but XHTML 1.0 Transitional declaration :(

File: Utils/View/View.php (hunk 1 above)
Comment (kojoty) [reply Utils/View/View.php]: ech - widzisz - niereformowalny ze mnie gość...

**Comment by kojoty** [comment]
> 1. OC PL and OC UK ready (have $config['ocNode'])

cool, thanks!

> 2. What is preffered $config['ocNode'] setting on devel site?

it's depends of what node you want to emulate...

> 3. Does this PR fix #958 and #1137?

sure :) - good to know - I've forgotten about these issues.

> 4. Can we remove "oc_on_all_pages_top_1A" and "oc_subtitle_on_all_pages_1A" from translation files?

I'm going to refactor translations in main layout at the end - it will be done. 

> 5. Can we remove nonlusers.txt after merge?

Yep. There is a change: there was two files in `<dyn-dir>/html/` now this is one json file in `<dyn-dir>.` I thinking if this is neccessary at all - this is a case of one sql query - I'm not sure if we need this at all..

> 6. Could you also change meta name="keywords" to new config value - in main.tpl.php mini.tpl.php

It will be done.

**Comment by andrixnet** [comment]
In the context of https://github.com/opencaching/opencaching-pl/issues/958 I believe `$config['ocNode']` is a poor choice. 
There already is `$oc_nodeid` which is numeric, which will be moved under $config (see my configuration refactor), with a different meaning and scope. 

`$config['ocNode']` does not mean **ocNode** but _node language_.

I had this in mind for the settings refactor: 
```
// Default language for the website. (2 letter language code)
$config['node']['lang']['language'] = 'pl'; 
// Default country. (2 letter country codee)
$config['node']['lang']['country'] = 'PL'; 
// Default locale setting
$config['node']['lang']['locale'] = 'pl-PL'; 
// Date format (PHP definition)
// FIXME ??? shouldn't this be taken from locale?
$config['node']['lang']['date_format'] = 'Y-m-d';
// Time format (PHP definition)
// FIXME ??? shouldn't this be taken from locale?
$config['node']['lang']['time_format'] = 'H:i';
// Date and Time format (PHP definition)
// FIXME ??? shouldn't this be taken from locale?
$config['node']['lang']['datetime_format'] = 'Y-m-d H:i'; 
```

with `$config['node']['lang']['language'] = 'pl';` having the same meaning as `$config['ocNode']`.

**Comment by andrixnet** [comment]
Nevertheless, `$config['ocNode'] = "ro";` set at OCRO in settings.inc.php

**Comment by kojoty** [comment]
@andrixnet,
I think this is a "deeper" problem of names for variables :)
I think that we both are partially right :)

You're right that we have $oc_nodeid, and you're right that is looks similar to language code.

But lets look at my example:
I want to allow site-menu configured per node - so I want to make files:
```
/Config/menu.default.php
/Config/menu.pl.php
/Config/menu.ro.php
/Config/menu.nl.php
...and so on
```
In the code I need to include one file - how to decide which one?
I need to get this node-identifier part like "pl" or "ro". Yes, I can create a simple translator: numericNodeId -> humanReadableNodeId but for me this is a wrong way.

Numbers are good in many places - but they have one disadvantage - people doesn't remember the meaning of numbers if they mean something different than just the number.

In my opinion $oc_nodeid encoding as a numbers was a poor decision. It is used in many places now so we will need to stay with it but it is difficult to understand for people (computer code is for people - not for computers!)

If we need numeric-node-id in the code we should translate it form human-readable form!
In config we should have "human-readable forms".

Then my `$config['ocNode']='pl'` means: "this is OCPL node". Yes, you right - it is looks like language symbol and can be a source of mistakes. Maybe it would be better to use something like for example
`$config['ocNode']="OCPL"` - but it is in my opinion too complicated. 

I believe `$config['ocNode']='pl'` is a good compromise. In future I think we should remove `$oc_nodeid` from config and just translate `$config['ocNode']='pl'` => 3 etc., and add another config switch for devel setups - but this is different discussion...

**Comment by kojoty** [comment]
@andrixnet, I better check the purpose of #958 and you're completely right - I didn't understand it when I wrote comment above - this is not a good fix for #958 - it should be fixed by `$config['node']['lang']['language']` - I will change it assume to your suggestions.

**Pull request merged**